### ISSUE TITLE
TypeError: environment can only contain strings

### ISSUE THREAD

**Issue by karthiknadig** [tox-dev/tox#1320]
## Environment:
* Pyhton: 2.7
* OS:  Windows 10
* tox: 3.12.0

## Output
```console
using tox.ini: D:\a\1\s\tox.ini (pid 2592)
using tox-3.12.0 from C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\__init__.pyc (pid 2592)
GLOB start: packaging 
GLOB sdist-make: D:\a\1\s\setup.py
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\__main__.py", line 4, in <module>
    tox.cmdline()
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\session\__init__.py", line 44, in cmdline
    main(args)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\session\__init__.py", line 68, in main
    exit_code = session.runcommand()
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\session\__init__.py", line 192, in runcommand
    return self.subcommand_test()
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\session\__init__.py", line 208, in subcommand_test
    venv.package = self.hook.tox_package(session=self, venv=venv)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\pluggy\hooks.py", line 289, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\pluggy\manager.py", line 68, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\pluggy\manager.py", line 62, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\pluggy\callers.py", line 208, in _multicall
    return outcome.get_result()
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\pluggy\callers.py", line 81, in get_result
    _reraise(*ex)  # noqa
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\pluggy\callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\package\__init__.py", line 16, in tox_package
    session.package, session.dist = get_package(session)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\package\__init__.py", line 29, in get_package
    package = acquire_package(config, session)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\package\__init__.py", line 40, in acquire_package
    path = build_package(config, session)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\package\builder\__init__.py", line 7, in build_package
    return make_sdist(config, session)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\package\builder\legacy.py", line 29, in make_sdist
    returnout=True,
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\site-packages\tox\action.py", line 93, in popen
    else 0
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\subprocess.py", line 390, in __init__
    errread, errwrite)
  File "C:\hostedtoolcache\windows\Python\2.7.14\x64\lib\subprocess.py", line 640, in _execute_child
    startupinfo)
TypeError: environment can only contain strings
```

## Repro steps:
1. clone https://github.com/microsoft/ptvsd.
2. Install packages in test_requirements.txt
3. Run tests via tox:
```
python -m tox -rvv -e py27 -- -vv 
```
Issue occurs with tox 3.12.0 and does not occur with 3.11.1

**Comment by karthiknadig** [comment]
Looks like this is likely fixed via https://github.com/tox-dev/tox/pull/1314 ?

**Comment by gaborbernat** [comment]
yeah just released as 3.12.1 https://pypi.org/project/tox/3.12.1/
